httpbakery/error.go

```diff
@@ -82,20 +84,68 @@ type ErrorInfo struct {
 	// macaroon.Signature() in hex).
 	CookieNameSuffix string `json:",omitempty"`
 
-	// VisitURL and WaitURL are associated with the
+	// The following fields are associated with the
 	// ErrInteractionRequired error code.
 
+	// InteractionMethods holds the set of methods that the
+	// third party supports for completing the discharge.
+	// See InteractionMethod for a more convenient
+	// accessor method.
+	InteractionMethods map[string]*json.RawMessage `json:",omitempty"`
+
 	// VisitURL holds a URL that the client should visit
 	// in a web browser to authenticate themselves.
+	// This is deprecated - it is superceded by the InteractionMethods
+	// field.
 	VisitURL string `json:",omitempty"`
 
 	// WaitURL holds a URL that the client should visit
 	// to acquire the discharge macaroon. A GET on
 	// this URL will block until the client has authenticated,
 	// and then it will return the discharge macaroon.
+	// This is deprecated - it is superceded by the InteractionMethods
+	// field.
 	WaitURL string `json:",omitempty"`
 }
 
+// SetInteraction sets the information for a particular
+// interaction kind to x.. The error should be an interaction-required
+// error. This method will panic if x cannot be JSON-marshaled.
+// In general, this method should not be used directly - higher
```

Comment:
Why shouldn't I use this? when is it safe to use it? Does that imply that all interaction methods have to be supported in the bakery?